src/module.c

```diff
@@ -357,15 +357,18 @@ typedef struct RedisModuleCommandFilterCtx {
     int argv_len;
     int argc;
     client *c;
+    int is_dirty;  /* indicates the data has been changed. */
```

Comment:
i don't see how the module can access that information.
furthermore, IIRC the plan was to expose an indication if the command succeeded or failed.
if we do all of that in `call`, we can use the same code that increments `failed_calls` for an indication (and we already have a `dirty` local variable too)

Reply:
Yes, if `moduleCallCommandPostFilters` been invoked in `call`, whether `dirty` local variable can be passed to the callback function?

Reply:
yes, but you added a flag in that struct, and didn't expose it to the module to access.
so the module callback should get two indications, one if the command succeeded or failed, and one if it modified the data or not.

Reply:
that context is an opaque struct (it is declared in module.c, not redismodule.h)
the module can't access it, and of course shouldn't (it wouldn't be ABI compatible with future changes).
point is, we should pass these as arguments for the callback, or add an additional API the module can use to extract it from the context.

Reply:
maybe it would do good if you'll try to add these capabilities to the test module. you'll quickly see if it works or not

Reply:
ohh, and also, you didn't handle the boolean indicating if the command succeeded or failed.

Reply:
Thank for your answer.

So, I should add an additional API to expose these two indications? for example:
`int (*RedisModule_CommandFilterCmdIsSucceeded)(RedisModuleCommandFilterCtx *fctx);`
`int (*RedisModule_CommandFilterDataIsDirty)(RedisModuleCommandFilterCtx *fctx);`